iceoryx_hoofs/test/moduletests/test_cxx_functional_interface_value_or.cpp

```diff
@@ -24,11 +24,15 @@ using namespace ::testing;
 constexpr bool TYPE_HAS_VALUE_METHOD = true;
 constexpr bool TYPE_HAS_NO_VALUE_METHOD = false;
 
+// the macro is used as code generator to make the tests more readable. because of the
+// template nature of those tests this is cannot be implemented in the same readable fashion
+// as with macros
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
 #define IOX_TEST_FUNCTIONAL_INTERFACE(TestName, variationPoint)                                                        \
     using SutType = typename TestFixture::TestFactoryType::Type;                                                       \
     constexpr bool HAS_VALUE_METHOD = iox::cxx::internal::HasValueMethod<SutType>::value;                              \
     ValueOrReturnsValueWhenValid<HAS_VALUE_METHOD>::template performTest<typename TestFixture::TestFactoryType>(       \
-        [](auto& sut, auto alternativeValue) { return variationPoint.value_or(alternativeValue); });
+        [](auto& sut, auto alternativeValue) { return (variationPoint).value_or(alternativeValue); });
```

Comment:
Why don't we need here `NOLINTNEXTLINE(bugprone-macro-parentheses)` as in the other files?

Reply:
Because we do not use this macro parameter. I removed it completely - an oversight